dotnet/src/MicroFocus.CafApi.QueryBuilder.Matcher/MatcherQueryVisitor.cs

```diff
@@ -41,145 +44,358 @@ public bool GetResult()
 
         public void VisitEquals(IMatcherFieldSpec<Document> fieldSpec, bool value)
         {
-            throw new NotImplementedException();
+            _isMatch = GetStringValues(fieldSpec).Any(v => bool.Parse(v) == value);
         }
 
         public void VisitEquals(IMatcherFieldSpec<Document> fieldSpec, long value)
         {
-            throw new NotImplementedException();
+            _isMatch = GetLongStream(fieldSpec).Any(v => v == value);
         }
 
         public void VisitEquals(IMatcherFieldSpec<Document> fieldSpec, string value)
         {
-            throw new NotImplementedException();
+            if (fieldSpec.IsCaseInsensitive)
+            {
+                _isMatch = GetFieldValuesAsStream(fieldSpec)
+                    .Any(v => string.Equals(v, value, StringComparison.CurrentCultureIgnoreCase));
```

Comment:
Using `StringComparison.CurrentCultureIgnoreCase` will cause different behavior on different platforms.  We should avoid anything that will make the code behave differently in different environments.